bin/compile

```diff
@@ -349,17 +347,29 @@ prune_devdependencies() {
 #   and we're using the default cache directories
 # then save off the cache after we prune out devDepenencies
 if [[ "$YARN" == "true" ]] && [[ $(features_get "cache-native-yarn-cache") == "true" ]] && [[ "$(get_cache_directories "$BUILD_DIR")" == "" ]]; then
-  meta_set "build-step" "prune-dependencies"
-  header "Pruning devDependencies" | output "$LOG_FILE"
-  prune_devdependencies | output "$LOG_FILE"
-  meta_set "build-step" "save-cache"
-  cache_build | output "$LOG_FILE"
+  if [[ "$YARN2" == "true" ]]; then
+    echo "Using Yarn2, writing to the cache and pruning of the node modules will be skipped."
+    meta_set "build-step" "prune-dependencies"
+    meta_set "build-step" "save-cache"
+  else
+    meta_set "build-step" "prune-dependencies"
+    header "Pruning devDependencies" | output "$LOG_FILE"
+    prune_devdependencies | output "$LOG_FILE"
+    meta_set "build-step" "save-cache"
+    cache_build | output "$LOG_FILE"
+  fi
 else
-  meta_set "build-step" "save-cache"
-  cache_build | output "$LOG_FILE"
-  meta_set "build-step" "prune-dependencies"
-  header "Pruning devDependencies" | output "$LOG_FILE"
-  prune_devdependencies | output "$LOG_FILE"
+  if [[ "$YARN2" == "true" ]]; then
+    echo "Using Yarn2, writing to the cache and pruning of the node modules will be skipped."
+    meta_set "build-step" "save-cache"
+    meta_set "build-step" "prune-dependencies"
+  else
+    meta_set "build-step" "save-cache"
+    cache_build | output "$LOG_FILE"
+    meta_set "build-step" "prune-dependencies"
+    header "Pruning devDependencies" | output "$LOG_FILE"
+    prune_devdependencies | output "$LOG_FILE"
+  fi
```

Comment:
Can do `if $YARN2; elif $YARN; else` here so that the Yarn 2 stuff doesn't have to be duplicated?

Also, we are referring to the package as `Yarn 2` in some places, and `Yarn2` in others. Do you mind changing this so that it says `Yarn 2`, and then making changes to the other `Yarn2`s also? (I did a global search, and there are only 3 + their respective tests)

Thank you!